pkg/component/controller/kuberouter.go

```diff
@@ -321,12 +322,25 @@ spec:
               fieldPath: metadata.name
         - name: KUBE_ROUTER_CNI_CONF_FILE
           value: /etc/cni/net.d/10-kuberouter.conflist
+        ports:
+        - name: healthz
+          containerPort: 20244
         livenessProbe:
           httpGet:
             path: /healthz
-            port: 20244
-          initialDelaySeconds: 10
+            port: healthz
+          initialDelaySeconds: 300
```

Comment:
Seems bit excessive? What's the reasoning for such a long delay?

Reply:
In my experience, liveness probes are only helpful in very few cases. Forcefully restarting a container over and over again is usually not helping much and will just increase churn/load on a cluster that is probably already busy with other things that lead to healthz answering with non-2xx responses. That's why I prefer high timeouts here. Henning [wrote a blog post about this](https://srcco.de/posts/kubernetes-liveness-probes-are-dangerous.html) back in the day.

Reply:
The basic difference here is that an app usually reports failures that it might be able to recover from by itself via the readiness endpoint. Restarting the app won't help and might even worse the situation in such a case. Unrecoverable errors should make an app terminate itself. This leaves the liveness probe to detect situations in which the app itself is broken due to things like deadlocks, tight endless loops, blocked on system calls that usually don't block for too long.

Reply:
👍